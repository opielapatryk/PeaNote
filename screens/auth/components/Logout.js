import React from 'react';
import {Pressable, Text,View} from 'react-native';
import { styles } from '../../../assets/styles/styles'
import { signOutAndClearReduxStore } from '../../../logic/funcMenu';
import { useDispatch, useSelector} from 'react-redux';
const Logout = ({navigation}) => {
    const { notes,pendingNotes } = useSelector((state)=>state.board)
    const dispatch = useDispatch()
    React.useEffect(() => {
        const unsubscribe = navigation.addListener('tabPress', (e) => {
          e.preventDefault();
          signOutAndClearReduxStore(notes,dispatch,pendingNotes)
        });
    
        return unsubscribe;
      }, [navigation]);
    
      return (
        <View>
          <Text>Logout!</Text>
        </View>
      );
}


export default Logout;
