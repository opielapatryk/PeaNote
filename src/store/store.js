import { configureStore } from '@reduxjs/toolkit'
import boardReducer from './notes/boardSlice'
import settingsReducer from './settings/settingsSlice'

export default configureStore({
    reducer:{
        board:boardReducer,
        settings:settingsReducer
    }
})