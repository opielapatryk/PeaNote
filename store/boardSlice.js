import { createSlice } from "@reduxjs/toolkit";

const initialState = {
    notes:[],
    isInput:true,   
    history:[]
}

export const boardSlice = createSlice({
    name:'board',
    initialState,
    reducers:{
        showInput: (state) =>{
            return {...state, isInput: true};
        },
        hideInput: (state) =>{
            return {...state, isInput: false};
        },
        addNote: (state, action) => {
            return {...state, notes:[...state.notes, action.payload],history: [...state.notes]}
        },
        removeNote: (state, action) => {
            return {
              ...state,
              notes: state.notes.filter(note => note.id !== action.payload),
              history: [...state.notes]
            };
          },
          changeInfo: (state, action) => {
            const prevText = state.history.find((note) => note.id === action.payload)?.text || "";
            return {
                ...state,
                notes: state.notes.map((note) => {
                    if (note.id === action.payload) {
                        return {
                            ...note,
                            isInfo: !note.isInfo,
                            text: note.isInfo ? prevText : 'click again to delete note',
                        };
                    }
                    return note;
                }),
                history: [...state.notes]
            };
        }     
    }
})

export const {showInput, addNote, removeNote, hideInput,changeInfo} = boardSlice.actions
export default boardSlice.reducer