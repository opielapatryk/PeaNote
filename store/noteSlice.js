import {createSlice } from '@reduxjs/toolkit'
const initialState = {
    id: 0,
    text:'',
    isNote:true,
    isInfo: false
}
export const noteSlice = createSlice({
    name:'note',
    initialState,
    reducers:{
        showInfo: (state)=>{
            return {...state,isInfo: true,text:'Click againt to delete note'}
        },
        setText: (state,action)=>{
            return {...state,text:action.payload}
        },
        hideInfo: (state)=>{
            return {...state,isInfo: false,text:initialState.text}
        },
        removeNote: (state)=>{
            return {...state,isNote: false}
        },
        isNoteTrue: (state)=>{
            return {...state,isNote: true}
        }
        
    }
})

export const {showInfo, hideInfo,removeNote,setText, isNoteTrue} = noteSlice.actions;

export default noteSlice.reducer