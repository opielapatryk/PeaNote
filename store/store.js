import { configureStore } from '@reduxjs/toolkit'
import boardReducer from './notes/boardSlice'

export default configureStore({
    reducer:{
        board:boardReducer
    }
})