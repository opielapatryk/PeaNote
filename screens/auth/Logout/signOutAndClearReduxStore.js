import {removeNote,removePendingNote} from '../../../store/notes/boardSlice';
import auth from '@react-native-firebase/auth';
import { GoogleSignin} from '@react-native-google-signin/google-signin';
import { setShowInput } from '../../../store/settings/settingsSlice';

export const signOutAndClearReduxStore = async (notes, dispatch, pendingNotes) => {
    try {
        dispatch(setShowInput(false))

        notes.forEach(sticker => dispatch(removeNote(sticker.id)));
        pendingNotes.forEach(sticker => dispatch(removePendingNote(sticker.id)));

        await auth().signOut().catch(err =>console.log(err))
        console.log("Successfully signed out");

        await GoogleSignin.revokeAccess().catch(err => console.log(err))
        console.log("Successfully revoked access");
    } catch (error) {
        console.error("Error during sign-out:", error);
        console.error("Google Sign-In Error Details:", error.toJSON());
    }
};
