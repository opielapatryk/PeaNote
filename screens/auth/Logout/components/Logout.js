import React from 'react';
import { signOutAndClearReduxStore } from '../functions/signOutAndClearReduxStore';
import { useDispatch } from 'react-redux';
import { View } from 'react-native'
import { styles } from '../../../../assets/styles/styles';
import Logo from '../../../../assets/images/logo.svg'
import LoginBody from '../../../public/Login/components/LoginBody';

const Logout = ({navigation}) => {
    const dispatch = useDispatch()

    React.useEffect(() => {
        const unsubscribe = navigation.addListener('focus', () => {
          signOutAndClearReduxStore(dispatch)
        });
        return ()=>{
          unsubscribe;
        }
      }, [navigation]);
    
      return (
        <View style={styles.container}>
          <Logo width={200} height={150}/>
          <LoginBody/>
        </View>
      );
}

export default Logout;