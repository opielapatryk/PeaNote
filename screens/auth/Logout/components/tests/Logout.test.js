import React from 'react';
import renderer from 'react-test-renderer';
import Logout from '../Logout';
import { NavigationContainer } from '@react-navigation/native';
import { act } from 'react-test-renderer';
import { Provider } from 'react-redux';  
import store from '../../../../../store/store'; 

  jest.mock('@react-native-firebase/firestore', () =>()=> {
    return {
      collection: jest.fn(()=>{
        return{
          where: jest.fn(()=>{
            return{
              get: jest.fn(() => Promise.resolve(true))
            }
          }),
        }
      }),
    };
  })

  jest.mock('@react-native-firebase/auth', () => ({
    auth: {
        signInWithEmailAndPassword: jest.fn(),
        createUserWithEmailAndPassword: jest.fn(),
        signOut: jest.fn(),
    },
  }));


jest.mock('@react-native-google-signin/google-signin', () => ({
  GoogleSignin: {
      signIn: jest.fn(),
  },
}));

test('Logout component renders correctly.', async () => {
    const navigation = {
      navigate: jest.fn(),
      addListener: jest.fn(),
    };

    const tree = renderer.create(
        <Provider store={store}>
            <NavigationContainer>
                <Logout navigation={navigation} />
            </NavigationContainer>
        </Provider>
      ).toJSON();

      await act(async () => {
        expect(tree).toMatchSnapshot();
      })
})
