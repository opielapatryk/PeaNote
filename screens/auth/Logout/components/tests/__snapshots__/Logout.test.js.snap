// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout component renders correctly. 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFDF3",
      "flex": 1,
      "justifyContent": "space-between",
      "paddingTop": 50,
    }
  }
>
  <logo.svg
    height={150}
    width={200}
  />
  <Text
    style={
      {
        "fontSize": 22,
      }
    }
  >
    Log In
  </Text>
  <Text
    style={
      {
        "color": "gray",
        "fontSize": 12,
      }
    }
  >
    Please provide your credentials by filling out the form below.
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Email"
    style={
      {
        "borderColor": "lightgrey",
        "borderRadius": 20,
        "borderWidth": 1,
        "marginBottom": 5,
        "marginTop": 5,
        "padding": 10,
        "width": 250,
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      {
        "borderColor": "lightgrey",
        "borderRadius": 20,
        "borderWidth": 1,
        "marginBottom": 5,
        "marginTop": 5,
        "padding": 10,
        "width": 250,
      }
    }
    value=""
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff2ba",
        "borderColor": "#e6c019",
        "borderRadius": 20,
        "borderWidth": 1,
        "elevation": 1,
        "marginBottom": 5,
        "marginTop": 5,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.18,
        "shadowRadius": 1,
        "width": 200,
      }
    }
  >
    <Text>
      Log In
    </Text>
  </View>
  <Text
    style={
      {
        "color": "gray",
        "fontSize": 12,
      }
    }
  >
    Or sign up with
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFDF3",
        "borderColor": "lightgray",
        "borderRadius": 20,
        "borderWidth": 1,
        "elevation": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 5,
        "marginTop": 5,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.18,
        "shadowRadius": 1,
        "width": 200,
      }
    }
  >
    <Text />
    <Text>
      Continue with Google
    </Text>
  </View>
  <View
    style={
      {
        "borderColor": "lightgray",
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "borderTopWidth": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFDF3",
          "borderColor": "lightgray",
          "borderRightWidth": 1,
          "elevation": 1,
          "marginBottom": 5,
          "marginTop": 5,
          "padding": 10,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.18,
          "shadowRadius": 1,
          "width": 150,
        }
      }
    >
      <Text>
        Create Account
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFDF3",
          "borderBottomColor": "#e6c019",
          "borderBottomLeftRadius": 30,
          "borderBottomRightRadius": 30,
          "borderBottomWidth": 4,
          "borderColor": "lightgray",
          "elevation": 1,
          "marginBottom": 5,
          "marginTop": 5,
          "padding": 10,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.18,
          "shadowRadius": 1,
          "width": 150,
        }
      }
    >
      <Text>
        Log In
      </Text>
    </View>
  </View>
</View>
`;
